### ISSUE TITLE
I tried running it once, but the effect wasn't as good. Can you help me take a look

### ISSUE THREAD

**Issue by favomj523** [ZombaSY/FSG-Net-pytorch#1]
![Image](https://github.com/user-attachments/assets/2f9a12d8-8c25-48b0-b386-37cdaea3e7e8)
Is it because I didn't use pre training>?

{
 ### Environment Parameters
  debug: false,
  mode: train,
  cuda: true,
  pin_memory: true,
  wandb: false, ## tsx change false -> true
  worker: 8,
  log_interval: 9999,
  save_interval: 1,
  saved_model_directory: 'model_ckpts',
  train_fold: 1,
  project_name: 'FSG-Net',
  CUDA_VISIBLE_DEVICES: '0',

  ### Train Parameters
  model_name: 'FSGNet',
    n_classes: 1,
    in_channels: 3,
  dataloader: 'Image2Image_zero_pad',
  criterion: 'DiceBCELoss', # DiceBCELoss
  task: 'segmentation',
  input_space: 'RGB',
  input_channel: 3,
  input_size: [608, 608],  # [h, w] DRIVE:608, STARE:704, CHASEDB1:1024, HRF:1344
  optimizer: 'AdamW',
    lr: 0.001,
    scheduler: 'WarmupCosine',  # ConstantLRSchedule, WarmupCosine
    cycles: 100,
    warmup_epoch: 20,
    weight_decay: 0.05, # 0.05
  batch_size: 8,
  epoch: 5000,
  ema_decay: 0, # set 0 to deactivate
  class_weight: [1.0, 1.0],
  model_path: '', # pretrained/FSGNet_DRIVE.pt
    freeze_layer: false,

  ### Augmentation Parameters
  transform_blur: true,
  transform_jitter: true,
  transform_hflip: true,
  transform_perspective: true,
  transform_cutmix: true,
  transform_rand_resize: true,
  transform_rand_crop: 288,

  train_x_path: './data/DRIVE/train/input',
  train_y_path: './data/DRIVE/train/label',
  val_x_path:   './data/DRIVE/val/input',
  val_y_path:   './data/DRIVE/val/label',
}

And i think the way to cal miou should be replace by this
iou = tp / (tp + fp + fn + epsilon) 
which have some wrong in u code
looking forward for you reply,thanks

**Comment by ZombaSY** [comment]
Dear @favomj523 ,

Actually it is class IoU which is not decribed on our paper.
The modified script is updated on new commit that is calculated on mean IoU.
Anyway, we fixed it even though not used anymore.

Please check release note [here!](https://github.com/ZombaSY/FSG-Net-pytorch/releases/tag/1.1.0)

**Comment by chenkanshigedang** [comment]
![Image](https://github.com/user-attachments/assets/bb46d749-3b88-4f24-8795-196a6cbfd3f3)
My reproduction results are slightly better than yours, but they still fail to reach the metrics in the author's weight file.

**Comment by ZombaSY** [comment]
@chenkanshigedang 
Could you reproduce the "inference.yaml"?

**Comment by chenkanshigedang** [comment]
> [@chenkanshigedang](https://github.com/chenkanshigedang) Could you reproduce the "inference.yaml"?

![Image](https://github.com/user-attachments/assets/d0a59702-5e5a-4718-a56d-f243476e41d5) Thank you for your reply. Here is my "inference.yaml".

**Comment by ZombaSY** [comment]
@chenkanshigedang 
Did you check the `model_path` value with proper weight?
You should use "FSG-Net-DRIVE.pt" on "DRIVE dataset", and so on.

fyi) Better to follow recent commit "e0b11355d67f735408a5724c5cc5ed736cb1234d"
